pwiz_tools/Skyline/SettingsUI/ViewLibraryDlg.cs

```diff
@@ -813,17 +828,40 @@ public void UpdateUI(bool selectionChanged = true)
                                 .GetSpectra(_peptides[index].Key, null, LibraryRedundancy.best).FirstOrDefault();
                         }
 
-                        var spectrumInfoR = LibraryRankedSpectrumInfo.NewLibraryRankedSpectrumInfo(spectrumInfo.SpectrumPeaksInfo,
-                                                                          transitionGroupDocNode.TransitionGroup.LabelType,
-                                                                          transitionGroupDocNode,
-                                                                          settings,
-                                                                          transitionGroupDocNode.Peptide.Target,
-                                                                          mods,
-                                                                          adducts,
-                                                                          types,
-                                                                          rankCharges,
-                                                                          rankTypes,
-                                                                          (spectrumInfo?.SpectrumHeaderInfo as BiblioSpecSpectrumHeaderInfo)?.Score);
+
+                        // Generates the object that will go into the property sheet

```

Comment:
Would be nice to put this into a function. This is already a very long function and you are adding a big chunk here that could be encapsulated to read:

_currentProperties = CreateSpectrumProperties(...);